### DISCUSSION TITLE
virus alert

### DISCUSSION THREAD

**Discussion by brianteeman** [mulaRahul/keyviz#11] [General]
I would love to install this but windows defender says that it contains a nasty virus Trojan:Script/Wacatac.B!ml and as a result it is blocked.

![image](https://user-images.githubusercontent.com/1296369/182348601-0295145a-8587-4835-b638-f8be2a77fa74.png)

**Comment by mulaRahul** [comment]
You're getting the message because **Keyviz** is signed with a personal certificate, not with a _Microsoft Partner Program_ account (I tried to sign up, but the site got stuck after step 1, don't know why...). The application reads **keyboard input**, which looks like a Trojan to the defender. You can use **Keyviz** by selecting "**Allow on device**" from the "**Actions**".

If you have doubts, you can check the whole source code or block it with [firewall](https://www.thewindowsclub.com/how-to-block-an-application-from-accessing-the-internet) to disable internet access for the same. No data is collected with the application so, it doesn't require an internet connection to work.

In the future, I'll try to tackle this either with a paid certificate or Microsoft Partner Program.

**Reply by brianteeman** [reply]
Checking the surce code doesn't help. The zip file is generated elsewhere and then added to github. So it could easily become infected.

**Reply by mulaRahul** [reply]
Yeah, you're right that a zip file can be infected. Though, if I add the .exe installer as it is, most probably you're browser will discard it as **harmful** software. You can see in this issue https://github.com/flutter/flutter/issues/95167, that apps build for windows with flutter already get detected as viruses/malware even if the app only displays just static text.

Of-course it's a bad user experience but currently, I'm unable to take further measures on the same. As I've mentioned earlier, I'll try to gather some more information on the same and find a solution. I hate to say it but you've to wait until then I guess. _Alternatively, you can get the source code, and build the app yourself._

**Comment by ThePharaohArt** [comment]
Checked on VirusTotal for the .exe installer v1.0.6, most likely false positives.
![image](https://github.com/mulaRahul/keyviz/assets/32932497/a9b77e75-dbc7-41e7-801e-049718219bca)

For the zip v1.0.6
![image](https://github.com/mulaRahul/keyviz/assets/32932497/45831872-a1ce-42b4-bd1e-d57033d66dce)

**Reply by sherkhan30452** [reply]
For portable version v1.0.6 (after extracting keyviz-v1.0.6-portable.zip ) Please see @mulaRahul 

![image](https://github.com/mulaRahul/keyviz/assets/33783004/e99033d7-4f1c-47b8-b1a3-d9f24ecfe658)

**Comment by mulaRahul** [comment]
Now you download Keyviz from [Microsoft Store](https://apps.microsoft.com/detail/Keyviz/9phzpj643p7l?mode=direct).

**Reply by kieukhuongthinh** [reply]
Updated in March 2nd, 2026. It says "This product is not currently available, try again later" in Microsoft Store.

<img width="1023" height="562" alt="image" src="https://github.com/user-attachments/assets/e43717bc-37e2-4300-b48e-bbf7c0b35003" />

**Reply by mulaRahul** [reply]
I'll perform a re-submission.
